check-vulnerabilities/action.yml

```diff
@@ -0,0 +1,390 @@
+name: "Check library and third-party vulnerabilities."
+description: >
+  Action actively checking for library and third party vulnerabilities by
+  means of the `bandit <https://bandit.readthedocs.io/en/latest/>`_ and
+  `safety <https://pyup.io/safety/>`_ Python packages. These packages
+  allow you to identify security issues and vulnerabilities inside your code.
+
+  This action has two running modes:
+  - **Default mode**: when running on default mode, if a security advisory
+    is detected, this action will publish both the security advisory and
+    an issue so that end users are informed about existing issues.
+  - **Development mode**: repository maintainers are asked to activate
+    the ``dev-mode`` option, which will activate both ``dry-run`` and
+    ``exit-with-error-on-new-advisory`` (independently of other configurations
+    provided). This way developers are informed of potential vulnerabilities
+    being introduced by them on their feature branch (while no new advisories
+    are created).
+
+
+inputs:
+  token:
+    description: >
+      Token with write permissions on the repository.
+    required: true
+    type: string
+
+  python-package-name:
+    description: >
+      Python package name being evaluate as it is shown on PyPI.
+    required: true
+    type: string
+
+  # Optional inputs
+
+  repo-full-name:
+    description: >
+      The repository to be evaluated. By default, it is extracted from the
+      GitHub context.
+    default: ''
+    required: false
+    type: string
+
+  dev-mode:
+    description: >
+      Whether to run or not this action in development mode. It will activate
+      by default the ``dry-run`` and ``exit-with-error-on-new-advisory`` flags.
+    required: false
+    default: false
+    type: boolean
+
+  dry-run:
+    description: >
+      Whether to run or not this action in dry run mode. Dry run mode executes
+      all the action steps and prints on screen the results (if any).
+    required: false
+    default: false
+    type: boolean
+
+  exit-with-error-on-new-advisory:
+    description: >
+      Whether to exit the action on error if a new advisory is detected.
+      This mode is not recommended on ``main`` branches since it will cause the
+      workflow to fail. To be used on ``feature`` branches in combination with
+      ``dry-run`` mode.
+    required: false
+    default: false
+    type: boolean
+
+  checkout:
+    description: >
+      Whether to clone the repository in the CI/CD machine. Default value is
+      ``true``.
+    default: true
+    required: false
+    type: boolean
+
+  python-version:
+    description: >
+      Desired Python version.
+    default: '3.10'
+    required: false
+    type: string
+
+runs:
+  using: "composite"
+  steps:
+    - name: "Declare needed environment variables"
+      shell: bash
+      run: |
+        echo "DEPENDENCY_CHECK_TOKEN=${{ inputs.token }}" >> $GITHUB_ENV

```

Comment:
I am worried about the secret nature of the token getting removed by copying its value to an environment variable.

Could we use "INPUT_TOKEN"? This one should be available, see https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#example-specifying-inputs

Reply:
Yep, definitely! :)